components/datadog/agent/docker.go

```diff
@@ -65,13 +65,15 @@ func dockerAgentComposeManifest(agentImagePath string, apiKey pulumi.StringInput
 			Version: "3.9",
 			Services: map[string]docker.ComposeManifestService{
 				"agent": {
+					Privileged:    true,
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
As the default agent config does not require to run with privileged rights, I suggest we have this as an option for when `system-probe` is enabled